packages/cache/src/internal/tar.ts

```diff
@@ -142,11 +144,18 @@ async function getArgs(
     tarPath.type === ArchiveToolType.BSD &&
     compressionMethod !== CompressionMethod.Gzip &&
     IS_WINDOWS
+
   if (BSD_TAR_ZSTD && type !== 'create') {
-    return [...compressionArgs, ...tarArgs].join(' ')
+    args = [...compressionArgs, ...tarArgs].join(' ')
   } else {
-    return [...tarArgs, ...compressionArgs].join(' ')
+    args = [...tarArgs, ...compressionArgs].join(' ')
+  }
+
+  if (BSD_TAR_ZSTD) {
+    args = ['cmd /c "', args, '"'].join(' ')
```

Comment:
Instead of using cmd, try if it is possible to run two exec commands as there is no pipe between `tar` and `zstd`.